cybersecurity/offensive/c2/apollo.yml

```diff
@@ -0,0 +1,746 @@
+    cmdline:
+      - cat
+      - -Path
+      - ${file}
+  cd:
+    description: Change working directory.
+    parameters:
+      dir:
+        type: string
+        description: The directory to change to.
+    cmdline:
+      - cd
+      - -Path
+      - ${dir}
+  cp:
+    description: Copy a file from path to destination.
+    parameters:
+      source:
+        type: string
+        description: The source file to copy.
+      destination:
+        type: string
+        description: The destination to copy the file to.
+    cmdline:
+      - cp
+      - -Path
+      - ${source}
+      - -Destination
+      - ${destination}
+  dcsync:
+    description: DCSync one or more user credentials.
+    parameters:
+      domain:
+        type: string
+        description: The domain to DCSync.
+      user:
+        type: string
+        description: The user to DCSync.
+      dc:
+        type: string
+        description: The domain controller to DCSync.
+    cmdline:
+      - dcsync
+      - -Domain
+      - ${domain}
+      - -User
+      - ${user}
+      - -DC
+      - ${dc}
+  download:
+    description: Download a file off the target system.
+    parameters:
+      Path:
+        type: string
+        description: The path to download the file to.
+      Host:
+        type: string
+        description: The hostname to download the file from.
+    cmdline:
+      - download
+      - -Path
+      - ${Path}
+      - -Host
+      - ${Host}
+  execute_assembly:
+    description: Execute a .NET assembly registered with register_file.
+    parameters:
+      assembly:
+        type: string
+        description: The assembly to execute.
+      args:
+        type: string
+        description: Arguments to pass to the assembly.
+    cmdline:
+      - execute_assembly
+      - -Assembly
+      - ${assembly}
+      - -Arguments
+      - ${args}
+  execute_coff:
+    description: Execute a object file (BOF) that's been registered with register_file.
+    parameters:
+      object:
+        type: string
+        description: The object file to execute.
+      function:
+        type: string
+        description: The function to execute.
+      timeout:
+        type: integer
+        description: The timeout for the execution.
+      args:
+        type: string
+        description: Arguments to pass to the object file.
+    cmdline:
+      - execute_coff
+      - -Coff
+      - ${object}
+      - -Function
+      - ${function}
+      - -Timeout
+      - ${timeout}
+      - -Arguments
+      - ${args}
+  execute_pe:
+    description: Execute a statically compiled executable that's been registered with register_file.
+    parameters:
+      binary:
+        type: string
+        description: The binary to execute.
+      args:
+        type: string
+        description: Arguments to pass to the binary.
+    cmdline:
+      - execute_pe
+      - -PE
+      - ${binary}
+      - -Arguments
+      - ${args}
+  exit:
+    description: Task agent to exit.
```

Comment:
```suggestion
    description: Task agent to exit.
    parameters:
      exit:
        type: boolean
        description: Exit the agent.
```

Reply:
in the validation, `parameters` is a required field so i'd suggest perhaps a boolean here :)

Reply:
[there's also a problem in my validation workflow](https://github.com/dreadnode/robopages/actions/runs/12062606645/job/33636519853) i'll look at fixing: (see [here](https://github.com/dreadnode/robopages/pull/27#pullrequestreview-2466387223), i think it may be a regression from #24)

```shell
Status: Downloaded newer image for dreadnode/robopages:latest
docker.io/dreadnode/robopages:latest
[2024-11-28T04:38:22Z ERROR] error while parsing "/workspace/apollo.yml": functions.exit: missing field `parameters` at line 161 column 5
```

Reply:
fixed the workflow in #28 and tested